### ISSUE TITLE
Lag and freezing when searching.

### ISSUE THREAD

**Issue by null-von-sushi** [rinigus/pure-maps#246]
Searching is incredibly laggy and freezes for a few seconds at times.

**Comment by rinigus** [comment]
Please specify platform, device, what do you search, which provider. Then I can ask for further details if needed

**Comment by null-von-sushi** [comment]
Sailfish OS Hossa on Xperia XA2. 
What do I search? Recently I tried (and failed) to find the train station in ifield (I searched for "ifield train station"). Not sure what you mean with provider, I was not aware it's possible to change what search to use?

**Comment by rinigus** [comment]
Depends on whether you use offline or online modes. Online providers are several. Note that offline mode is not working in 3.0.3, see talk.maemo.org thread of OSM Scout Server for reasons and when to expect support

**Comment by null-von-sushi** [comment]
Okay, for some reason I cannot reproduce it myself anymore, sorry.
If it helps, this is what happened in chronological order:

- No internet, try to search. Doesn't work (offline scout server doesn't seem to do search?). Also no GPS fix.
- Connect to internet, try to search.
- About 7 minutes pass
- Find postcode on google maps website, search for it on pure maps
- application takes a good 10 seconds to go from frozen to lagging but kinda useable
- Another ~5 minutes, I finally get GPS lock. 

I now installed `espeak` and configured location settings to use mozilla backend too.

**Comment by null-von-sushi** [comment]
I set pure maps to 'mixed' and try to use offline. I think it did work for me, kinda? At least I can see the region I downloaded in OSM Scout Server (and a bit more, or the region is just larger then I thought) in Pure Maps. I can even zoom and stuff (this is with airplane mode on).

**Comment by rinigus** [comment]
Until fully supported osm scout server will come out for 3.0.3.x, don't rely on it for much. It may work partially, but I will need to recompile it for 3.0.3 at OBS. So, stick with online mode for now and test with that

**Comment by rinigus** [comment]
As for TTS, I would recommend mimic or picoTTS. espeak is very bad.

ZLib versions seem to mismatch which may ruin the server - hence the waiting

**Comment by null-von-sushi** [comment]
I think I tried mimic on actual linux? That's the mycroft one isn't it? Anyways, it didn't wanna install (using storeman) which is why I went for speak. I'll just download the RPM manually. I don't really use voice navigation, I just wanted to install it in case I did need it (and yes espeak is pretty awful). 

I'd use online mode, except that I don't have internet in the car :P 

But thanks, I guess my report is a bit vague/unreproducible and quite possibly the result of an error with OSM Scout Server, so feel free to close the issue as I am not sure what else could be dine.
